### PR TITLE
Provide functions to unwrap value or error

### PR DESCRIPTION
**Pull request by CommanderTvis**
closes: #76

### CODE REVIEW

File: result/src/test/kotlin/com/github/kittinunf/result/ResultTests.kt [hunk 1]
```diff
@@ -2,11 +2,9 @@ package com.github.kittinunf.result
 
 import kotlinx.coroutines.delay
 import kotlinx.coroutines.runBlocking
-import org.hamcrest.CoreMatchers.equalTo
-import org.hamcrest.CoreMatchers.instanceOf
-import org.hamcrest.CoreMatchers.notNullValue
-import org.hamcrest.CoreMatchers.nullValue
+import org.hamcrest.CoreMatchers.*
```
Comment (kittinunf) [inline result/src/test/kotlin/com/github/kittinunf/result/ResultTests.kt:5]: Ideally we should have this in the git tracking `.idea/`, but would you mind revert this back to non star version?

File: result/src/main/kotlin/com/github/kittinunf/result/Result.kt [hunk 2]
```diff
@@ -112,6 +112,13 @@ inline fun <V : Any?, reified E : Exception> List<Result<V, E>>.lift(): Result<L
     }
 }
 
+inline fun <V, E : Exception> Result<V, E>.unwrap(failure: (E) -> Nothing): V =
```
Comment (kittinunf) [inline result/src/main/kotlin/com/github/kittinunf/result/Result.kt:115]: I am having hard time understand why this is better than the current API that we already have?

File: result/src/test/kotlin/com/github/kittinunf/result/ResultTests.kt (hunk 1 above)
Comment (CommanderTvis) [reply result/src/test/kotlin/com/github/kittinunf/result/ResultTests.kt:5]: OK

File: result/src/main/kotlin/com/github/kittinunf/result/Result.kt (hunk 2 above)
Comment (CommanderTvis) [reply result/src/main/kotlin/com/github/kittinunf/result/Result.kt:115]: How can I do:
```kotlin
val myValue = result.unwrap { throw RuntimeException(it) }
```

Comment (CommanderTvis) [reply result/src/main/kotlin/com/github/kittinunf/result/Result.kt:115]: @kittinunf

**Review by kittinunf — APPROVED** [review]

**Comment by kittinunf** [comment]
I think this is a good addition to the library! 👍

**Pull request merged**